### PR TITLE
[Keyboard] Add amigopunk, a handwired 75% ALPS

### PR DESCRIPTION
**Pull request by haesbaert**
Add a new handwired keyboard made by me.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/amigopunk/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE ?= yes  # Audio control and System control(+450)
+CONSOLE_ENABLE ?= no    # Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes      # USB Nkey Rollover
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no      # Enable audio
+RGBLIGHT_ENABLE ?= no   # Enable RGB
+ENCODER_ENABLE ?= yes   # Enable rotary encoders
+OLED_DRIVER_ENABLE ?= yes # OLED screen
```
Comment (drashna) [inline keyboards/handwired/amigopunk/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = yes # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes  # Mouse keys
EXTRAKEY_ENABLE = yes  # Audio control and System control
CONSOLE_ENABLE = no    # Console for debug
COMMAND_ENABLE = no    # Commands for debug and configuration
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes      # USB Nkey Rollover
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
AUDIO_ENABLE = no      # Enable audio
RGBLIGHT_ENABLE = no   # Enable RGB
ENCODER_ENABLE = yes   # Enable rotary encoders
OLED_DRIVER_ENABLE = yes # OLED screen
```

File: keyboards/handwired/amigopunk/config.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Encoder setup */
+#ifdef ENCODER_ENABLE
+#define ENCODERS_PAD_A { E0 }
+#define ENCODERS_PAD_B { E1 }
+#define ENCODER_RESOLUTION 4
```
Comment (drashna) [inline keyboards/handwired/amigopunk/config.h:45]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by haesbaert** [comment]
I've commited the requested changes.

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/amigopunk/config.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x1805
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    haesbaert
+#define PRODUCT         amigopunk
+#define DESCRIPTION     keyboard
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/config.h:27]:
```suggestion
```

File: keyboards/handwired/amigopunk/info.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                {"label":"K57 (C5,E7)", "x":4, "y":5.25, "w":7},
+                {"label":"K5B (C5,F2)", "x":11, "y":5.25, "w":1.5},
+                {"label":"K5D (C5,F4)", "x":12.5, "y":5.25, "w":1.5},
+                {"label":"K5E (C5,F5)", "x":14, "y":5.25},
+                {"label":"K5F (C5,F6)", "x":15, "y":5.25},
+                {"label":"K5G (C5,F7)", "x":16, "y":5.25}
+            ]
+        }
+    }
+    ,"meta": "https://noroadsleft.github.io/kbf_qmk_converter/"
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/info.json:93]:
```suggestion
```

File: keyboards/handwired/amigopunk/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+static void render_qmk_logo(void) {
+    static const char PROGMEM qmk_logo[] = {
+        0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8A, 0x8B, 0x8C, 0x8D, 0x8E, 0x8F, 0x90, 0x91, 0x92, 0x93, 0x94,
+        0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xB2, 0xB3, 0xB4,
+        0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0x00
+    };
+
+    oled_write_P(qmk_logo, false);
+}
+#endif	/* if 0 */
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/keymaps/default/keymap.c:59]:
```suggestion
```

File: keyboards/handwired/amigopunk/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        0x0c, 0x07, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+    };
+    oled_write_raw_P(amigopunk_logo, sizeof(amigopunk_logo));
+}
+void oled_task_user(void) {
+    /* render_qmk_logo(); */
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/keymaps/default/keymap.c:99]:
```suggestion
```

File: keyboards/handwired/amigopunk/readme.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# amigopunk
+
+![amigopunk](https://imgur.com/a/9Z37fWz)
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/readme.md:3]:
```suggestion
![amigopunk](https://i.imgur.com/7uVsX1vh.jpeg)
```

File: keyboards/handwired/amigopunk/rules.mk [hunk 8]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = at90usb1286
+
+# Bootloader for a teensy 2.0++
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/rules.mk:4]:
```suggestion
# Bootloader selection
```

File: keyboards/handwired/amigopunk/rules.mk [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes  # Audio control and System control
+CONSOLE_ENABLE = no    # Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes      # USB Nkey Rollover
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no      # Enable audio
+RGBLIGHT_ENABLE = no   # Enable RGB
+ENCODER_ENABLE = yes   # Enable rotary encoders
+OLED_DRIVER_ENABLE = yes # OLED screen
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/rules.mk:21]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
OLED_DRIVER_ENABLE = yes
```

File: keyboards/handwired/amigopunk/readme.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# amigopunk
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/readme.md:1]: This file should follow the [template](https://github.com/qmk/qmk_firmware/blob/master/quantum/template/avr/readme.md).

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/amigopunk/readme.md (hunk 10 above)
Comment (haesbaert) [reply keyboards/handwired/amigopunk/readme.md:1]: Done, leme know if I should squash everything.

File: keyboards/handwired/amigopunk/readme.md [hunk 11]
```diff
@@ -1,7 +1,19 @@
 # amigopunk
 
-![amigopunk](https://i.imgur.com/7uVsX1vh.jpeg)
+[amigopunk](https://i.imgur.com/7uVsX1vh.jpeg)
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/readme.md:3]:
```suggestion
![amigopunk](https://i.imgur.com/7uVsX1vh.jpeg)
```

File: keyboards/handwired/amigopunk/readme.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+# amigopunk
+
+[amigopunk](https://i.imgur.com/7uVsX1vh.jpeg)
+
+A 75% handwired with rotary encoder and OLED using matias clicky ALPS.
+
+Keyboard Maintainer: [haesbaert](https://github.com/haesbaert)
+Hardware Supported: Teensy 2.0++
+Hardware Availability: haesbaert@haesbaert.org
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/readme.md:9]:
```suggestion
* Keyboard Maintainer: [haesbaert](https://github.com/haesbaert)
* Hardware Supported: Teensy 2.0++
* Hardware Availability: haesbaert@haesbaert.org
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/amigopunk/readme.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+* Hardware Supported: Teensy 2.0++
+* Hardware Availability: haesbaert@haesbaert.org
+
+Make example for this keyboard (after setting up your build environment):
+
+    make amigopunk:default
+
+Flashing example for this keyboard:
+
+    make amigopunk:default:flash
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/readme.md:17]:
```suggestion
    make handwired/amigopunk:default

Flashing example for this keyboard:

    make handwired/amigopunk:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/amigopunk/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+    ),
+
+};
+
+#ifdef ENCODER_ENABLE
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index != 0)
+        return;
+
+    clockwise ? tap_code(KC_VOLU) : tap_code(KC_VOLD);
```
Comment (fauxpark) [inline keyboards/handwired/amigopunk/keymaps/default/keymap.c:44]:
```suggestion
    tap_code(clockwise ? KC_VOLU : KC_VOLD);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/amigopunk/keymaps/default/keymap.c (hunk 14 above)
Comment (haesbaert) [reply keyboards/handwired/amigopunk/keymaps/default/keymap.c:44]: No, this is bikeshedding.

Comment (fauxpark) [reply keyboards/handwired/amigopunk/keymaps/default/keymap.c:44]: Not really...

Comment (haesbaert) [reply keyboards/handwired/amigopunk/keymaps/default/keymap.c:44]: sorry, but it is :).
I'm not changing that.

Comment (haesbaert) [reply keyboards/handwired/amigopunk/keymaps/default/keymap.c]: ok have at it, it's your repo.
I commited the change

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**